### PR TITLE
Wrapping up possible crashes

### PR DESCRIPTION
**Pull request by phoinic**

### CODE REVIEW

File: src/ipa_pc/mod.rs [hunk 1]
```diff
@@ -69,9 +71,9 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
 
     /// Complete semantic checks on `ck`.
     #[inline]
-    pub fn check_key(ck: &CommitterKey<G>, max_degree: usize) -> bool {
-        let pp = <Self as PolynomialCommitment<G::ScalarField>>::setup(max_degree).unwrap();
-        ck.is_valid() && &pp.hash == &ck.hash
+    pub fn check_key(ck: &CommitterKey<G>, max_degree: usize) -> Result<bool, Error> {
```
Comment (DanieleDiBenedetto) [inline src/ipa_pc/mod.rs:74]: setup returns a Result for the purpose of being generic in the trait. Here we know that we are in poly commit and the setup cannot fail, so the unwrap was ok. let's restore as it was before

File: src/ipa_pc/mod.rs [hunk 2]
```diff
@@ -275,10 +282,10 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
                 if (degree_bound_len + 1) % key_len != 0 { Some(degree_bound_len + 1) } else { None }
             });
 
-            assert_eq!(
-                degree_bound_len.is_some(),
-                commitment.shifted_comm.is_some()
-            );
+            // TODO: check error message
+            if degree_bound_len.is_some() != commitment.shifted_comm.is_some() {
+                Err(Error::Other("Degree bound and shifted commitment are not synced".to_owned()))?
```
Comment (DanieleDiBenedetto) [inline src/ipa_pc/mod.rs:287]: "Degree bound and shifted commitment must be both either present or not present"

File: src/ipa_pc/mod.rs [hunk 3]
```diff
@@ -300,7 +307,11 @@ impl<G: AffineCurve, D: Digest> InnerProductArgPC<G, D> {
             }
         }
 
-        assert_eq!(proof.hiding_comm.is_some(), proof.rand.is_some());
+        // TODO: check error message
```
Comment (DanieleDiBenedetto) [inline src/ipa_pc/mod.rs:310]: "Hiding commitment and proof randomness must be both either present or not present"

File: src/ipa_pc/mod.rs [hunk 4]
```diff
@@ -876,7 +897,13 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
         polys_iter.zip(comms_iter.zip(rands_iter))
         {
             let label = labeled_polynomial.label();
-            assert_eq!(labeled_polynomial.label(), labeled_commitment.label());
+            if labeled_polynomial.label() != labeled_commitment.label() {
+                Err(Error::Other(format!(
+                    "Labels ara not equal: poly '{}'. commitment '{}'",
```
Comment (DanieleDiBenedetto) [inline src/ipa_pc/mod.rs:902]: typo ara -> are

**Review by DanieleDiBenedetto — CHANGES_REQUESTED** [review]

**Review by DanieleDiBenedetto — APPROVED** [review]
Fine, remember to  remove the remainings TODO: Check error message

**Review by phoinic — COMMENTED** [review]
> Fine, remember to remove the remainings TODO: Check error message
Done

**Pull request merged**